R/layer.R

```diff
@@ -292,15 +292,23 @@ Layer <- ggproto("Layer", NULL,
     }
 
     n <- nrow(data)
+    aes_n <- lengths(evaled)
     if (n == 0) {
       # No data, so look at longest evaluated aesthetic
       if (length(evaled) == 0) {
         n <- 0
       } else {
-        aes_n <- lengths(evaled)
         n <- if (min(aes_n) == 0) 0L else max(aes_n)
       }
     }
+    if ((self$geom$check_constant_aes %||% TRUE)
+        && length(aes_n) > 0 && all(aes_n == 1) && n > 1) {
+      cli::cli_warn(c(
+        paste0("All aesthetics have length 1, but the data has {n} rows ",
+               "in {.fn {self$constructor[1]}}."),
```

Comment:
Not sure if `self$constructor[1]` is a very safe pattern to report the layer

Reply:
See https://github.com/tidyverse/ggplot2/blob/d7f22413efea3dd2a7c9effff05d4b2aa2c2d300/R/plot-build.R#L430

We prefer to pass this in as the surrounding call, rather than name it in the error text

Reply:
Thanks that makes much more sense!